### PR TITLE
Popup and Overlay: move drag and resize logic from Overlay to Popup

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

File: js/ui/popup/ui.popup.js [hunk 1]
```diff
@@ -661,6 +741,8 @@ const Popup = Overlay.inherit({
     },
 
     _optionChanged: function(args) {
```
Comment (ksercs) [inline js/ui/popup/ui.popup.js:743]: lets use destructurization here and get rid of `args` using in this function

**Review by ksercs — DISMISSED** [review]

File: js/ui/popup/ui.popup.js (hunk 1 above)
Comment (Zedwag) [reply js/ui/popup/ui.popup.js:743]: looks like we shouldn't do this, because we pass `args` to the `callBase()` function

**Review by DokaRus — DISMISSED** [review]

**Review by ksercs — DISMISSED** [review]

File: js/ui/popover.d.ts [hunk 2]
```diff
@@ -43,6 +48,15 @@ export type HiddenEvent = EventInfo<dxPopover>;
...
 
+/** @hidden */
+export type ResizeEvent = NativeEventInfo<dxPopup, MouseEvent | TouchEvent> & ResizeInfo;
+
+/** @hidden */
+export type ResizeStartEvent = NativeEventInfo<dxPopup, MouseEvent | TouchEvent> & ResizeInfo;
+
+/** @hidden */
+export type ResizeEndEvent = NativeEventInfo<dxPopup, MouseEvent | TouchEvent> & ResizeInfo;
+
```
Comment (IlyaKhD) [inline js/ui/popover.d.ts]:
```suggestion
```
Popover doesn't have these events

File: js/ui/popover.d.ts [hunk 3]
```diff
@@ -75,6 +89,22 @@ export interface dxPopoverOptions<TComponent> extends dxPopupOptions<TComponent>
...
+     * @docid
+     * @default false
+     * @hidden
+     */
+    dragEnabled?: boolean;
+    /**
+     * @docid
+     * @hidden
+     */
+    dragOutsideBoundary?: boolean;
```
Comment (IlyaKhD) [inline js/ui/popover.d.ts]:
```suggestion
```
Popover doesn't have these props

File: js/ui/popover.d.ts [hunk 4]
```diff
@@ -98,13 +128,61 @@ export interface dxPopoverOptions<TComponent> extends dxPopupOptions<TComponent>
...
+     * @type_function_param1_field1 component:this
+     * @type_function_param1_field2 element:DxElement
+     * @type_function_param1_field3 model:any
+     * @type_function_param1_field4 event:event
+     * @type_function_param1_field5 width:number
+     * @type_function_param1_field6 height:number
+     * @action
+     * @hidden
+     */
+    onResizeStart?: ((e: ResizeStartEvent) => void);
```
Comment (IlyaKhD) [inline js/ui/popover.d.ts]:
```suggestion
```

File: js/ui/popover.d.ts [hunk 5]
```diff
@@ -98,13 +128,61 @@ export interface dxPopoverOptions<TComponent> extends dxPopupOptions<TComponent>
...
      * @type Enums.Position|PositionConfig
      * @default { my: 'top center', at: 'bottom center', collision: 'fit flip' }
      * @public
      */
     position?: 'bottom' | 'left' | 'right' | 'top' | PositionConfig;
+    /**
+     * @docid
+     * @hidden
+     */
+    resizeEnabled?: boolean;
```
Comment (IlyaKhD) [inline js/ui/popover.d.ts]:
```suggestion
```

**Comment by IlyaKhD** [comment]
Please address my comments before merging

**Comment by Zedwag** [comment]
> Please address my comments before merging

Sure, now I'm looking for another solution, like deviding Popup into Popup and PupupFull (where we can hide options we want to hide for Popover)

File: js/ui/popover.d.ts (hunk 2 above)
Comment (Zedwag) [reply js/ui/popover.d.ts]: all done

**Review by ksercs — DISMISSED** [review]

**Review by ksercs — DISMISSED** [review]

**Review by DokaRus — DISMISSED** [review]

File: js/ui/popup.d.ts [hunk 6]
```diff
@@ -90,13 +90,25 @@ export interface dxPopupOptions<TComponent> extends dxOverlayOptions<TComponent>
      * @public
      */
     container?: string | UserDefinedElement;
+    /**
+     * @docid
+     * @default undefined
+     * @public
+     */
+     dragAndResizeArea?: string | UserDefinedElement;
```
Comment (IlyaKhD) [inline js/ui/popup.d.ts]:
```suggestion
    dragAndResizeArea?: string | UserDefinedElement;
```

File: js/ui/popup.d.ts [hunk 7]
```diff
@@ -90,13 +90,25 @@ export interface dxPopupOptions<TComponent> extends dxOverlayOptions<TComponent>
...
      * @default true &for(desktop)
      * @public
      */
     dragEnabled?: boolean;
+     /**
+     * @docid
+     * @default false
+     * @public
+     */
+      dragOutsideBoundary?: boolean;
```
Comment (IlyaKhD) [inline js/ui/popup.d.ts]:
```suggestion
     dragOutsideBoundary?: boolean;
```

**Review by ksercs — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Pull request merged**